### PR TITLE
Remove Major/Minor from polaris-icons

### LINKED ISSUE #10813: [Icons] Merge and migrate icons so we have a single unified set
## Issue summary
Currently we still have the legacy concept of `Major` and `Minor` in our icon naming even though post uplift, all icons are the same size. This means we have duplicate icons in our icon library. This bloats our icon library and produces extra maintenance cost because the duplicated assets have to be kept in sync over time. 

Here you can see all the visual duplicated icons in our library that we could eliminate (80+ icons).

Screenshot 2023-10-03 at 12 11 59 PM

We need to run migrations on our library so we have a single unified set

1. Remove all `Major` and `Minor` from any icon names
2. When an icon has both a `Major` and `Mino`r variation, point them to the same asset
3. Deprecate/delete all duplicates so we only a single asset

### PR DESCRIPTION
**Pull request by sam-b-rose**
Closes #10813 

Removes the Major and Minor suffix from all icons. Support backwards compatibility without duplicating assets.

You can see the full list of icon SVG assets here:

https://unpkg.com/browse/@shopify/polaris-icons@0.0.0-snapshot-release-20231102192017/dist/svg/

### CODE REVIEW

File: .changeset/odd-beans-check.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': minor
```
Comment (kyledurand) [inline .changeset/odd-beans-check.md:2]:
```suggestion
'@shopify/polaris-icons': major
```

Comment (sam-b-rose) [reply .changeset/odd-beans-check.md:2]: Was going to first make this backwards compatible. Then quickly follow up with a major

**Review by zecarlostorre — APPROVED** [review]
Thanks for this! We'll update the Figma libraries accordingly.

**Review by heyjoethomas — APPROVED** [review]
Absolute legends 🚀

**Pull request merged**